### PR TITLE
Basic workflow for basecalling ONT pod5 files. Does not yet include demultiplexing.

### PR DESCRIPTION
**Pull request by simonleandergrimm**
This PR creates the necessary files to basecall Oxford Nanopore pod5 files. Running the basecaller (dorado) requires a g5.xlarge machine, with an AMI that is both ECS-optimized and contains GPU-drivers ([link](https://github.com/aws/amazon-ecs-ami/releases)). The Instance template can be found [here](https://us-east-1.console.aws.amazon.com/ec2/home?region=us-east-1#LaunchTemplateDetails:launchTemplateId=lt-06b73c764967b6710)

@willbradshaw I had to specify default values for undefined run.nf parameters as I always got warnings that they were not defined (even though I wasn't running run.nf). I will resolve this in future, though let me know if I can suppress these warnings in some other way.

Also, let me know if there is a more efficient way to get the base_dir leaf, which I'd need to define the name of the nanopore_run:
```
params {
    mode = "basecall"
    debug = true

    // Directories
    base_dir = "s3://nao-mgs-simon/NAO-ONT-20240905-DCS_RNA2_Test"

    // Run parameters
    // TODO: Automate getting run name from base_dir
    nanopore_run = "NAO-ONT-20240905-DCS_RNA2_Test"
```

### CODE REVIEW

**Comment by willbradshaw** [comment]
> @willbradshaw I had to specify default values for undefined run.nf parameters as I always got warnings that they were not defined (even though I wasn't running run.nf. I will resolve this in future, though let me know if I can suppress these warnings in some other way.

Yeah this is a known issue that someone (possibly me) should look into fixing. If we're going to have multiple workflows running from the same start point this is going to keep coming up.

> Also, let me know if there is a more efficient way to get the base_dir leaf, which I'd need to define the name of the nanopore_run

Sorry, I don't understand what you mean by this.

**Comment by simonleandergrimm** [comment]
> > Also, let me know if there is a more efficient way to get the base_dir leaf, which I'd need to define the name of the nanopore_run
> 
> 
> 
> Sorry, I don't understand what you mean by this.


> > Also, let me know if there is a more efficient way to get the base_dir leaf, which I'd need to define the name of the nanopore_run
> 
> 
> 
> Sorry, I don't understand what you mean by this.

When the s3 address is s3://foo/bar, I want the nanopore run identifier to be bar. After trying for 5 min I didn't figure out how to do this.

File: bin/batch_pod5.py
Comment (willbradshaw) [inline bin/batch_pod5.py:1]: @simonleandergrimm can you put this in the appropriate module directory? We're using module binaries now.

File: configs/basecall.config [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    base_dir = "s3://nao-mgs-simon/NAO-ONT-20240905-DCS_RNA2_Test"
+
+    // Run parameters
+    // TODO: Automate getting run name from base_dir
+    nanopore_run = "NAO-ONT-20240905-DCS_RNA2_Test"
+
+    // TODO: Add optional paramter for ONT kit
+
+    // Files
+    pod5_dir = "${base_dir}/pod5_small/"
```
Comment (willbradshaw) [inline configs/basecall.config:19]: Do these need to be in the config file? If they'll always be the same except for the base_dir you can hardcode them in the pipeline and save the user some thinking.

File: configs/basecall.config [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    bam_dir = "${base_dir}/bam/"
+    calls_bam = "${base_dir}/bam/calls.bam"
+    fastq_file = "${base_dir}/raw/${nanopore_run}.fastq.gz"
+
+    // Default values for undefined run.nf parameters. Adding them here because they always get triggered
```
Comment (willbradshaw) [inline configs/basecall.config:24]: I _think_ this should be fixed now that I've removed all the addParams statements. Try it again after merging in the changes from master?

File: configs/basecall.config [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    quality_encoding = "auto"
+    kraken_memory = "8G"
+    bbduk_suffix = ""
+    adapters = ""
+}
+
+
+includeConfig "${projectDir}/configs/containers.config"
+includeConfig "${projectDir}/configs/resources.config"
+includeConfig "${projectDir}/configs/profiles.config"
```
Comment (willbradshaw) [inline configs/basecall.config:36]: Some missing standard configs to add once you've merged in master.

File: configs/basecall.config [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+params {
+    mode = "basecall"
+    debug = true
+
+    // Directories
+    base_dir = "s3://nao-mgs-simon/NAO-ONT-20240905-DCS_RNA2_Test"
+
+    // Run parameters
+    // TODO: Automate getting run name from base_dir
+    nanopore_run = "NAO-ONT-20240905-DCS_RNA2_Test"
```
Comment (willbradshaw) [inline configs/basecall.config:14]: If this value is always equivalent to the basename of base_dir, you can do this with Groovy: https://www.nextflow.io/docs/latest/working-with-files.html#getting-file-attributes

File: configs/containers.config [hunk 5]
```diff
@@ -63,4 +63,12 @@ process {
     withLabel: fastp {
         container = "staphb/fastp:0.23.4"
     }
+    withLabel: dorado {
+        container = "ontresearch/dorado:latest"
+        // NB: For now going with latest version, maybe the version switching with new updates will break things in the future.
```
Comment (willbradshaw) [inline configs/containers.config:68]: I would recommend locking in a version soon

File: workflows/basecall.nf [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+        // Demultiplexing
+        demux_ch = DEMUX_POD_5(bam_ch.bam, params.kit)
+
+        // Convert to FASTQ
+        fastq_ch = BAM_TO_FASTQ(demux_ch.demux_bam, params.nanopore_run)
+
+
+
+    publish:
```
Comment (willbradshaw) [inline workflows/basecall.nf:43]: These need to be compatible with the output structure specified in `main.nf` (@simonleandergrimm look into this more after merging in master)

**Pull request closed without merging**